module4-authentication-and-security/r1.1.1-authentication-persistence/README.md

```diff
@@ -0,0 +1,268 @@
+# Authentication Persistence
+
+In this lesson, we will tackle how to persist user login, or registration between requests. Learning objectives are:
+
+- Difference between stateful and stateless persistence
+- Implementing stateful persistence using express-sessions
+- Implementing stateless identifiers using JSON Web Tokens (JWT)
+
+## **Stateful Session Storage**
+
+In web servers, sessions are server-side temporary storage tools. They are used for a variety of use cases to store temporary data between requests. They use different storage adapters, like RAM, disk, databases, in-memory cache, etc. to save the temporary data.
+
+They are different than database data persistent in that session data are simple key-value stores that can be invalidated and become redundant easily. They are some handy copies (cache) of the original data from the database that are supposedly easier and faster to access and deleting them won't affect the server.
+
+In Express.js, we can use the library [express-session](https://www.npmjs.com/package/express-session) to create session storage. It can be plugged into the Express app as a middleware that attaches a `session` object to any request that comes to the server.
+
+```js
+// ./app.js
+
+const express = require("express");
+const sessions = require("express-session");
+
+// Init our express app
+const app = express();
+
+// Below is where we introduce encrypted sessions
+// Make sure your app secret is unique and
+// defined by cryptographic random generator.
+// It has in-built ability to parse and use cookies
+app.use(
+  sessions({
+    secret: "MY_APP_SECRET",
+    cookie: { secure: false }, // make sure to change this to true on production code
+  })
+);
+
+let listener = app.listen(5000, function () {
+  console.log("Listening on port " + listener.address().port);
+});
+```
+
+In the code snippet above, we configured the session storage using `app.use()` and created a new session instance with the configuration below:
+
+- `secret`: a token that will be used to encrypt the data and cookies. It can also be a universal `APP_SECRET` defined in the environment variables. This must be a random secret generated by secure random generators.
+- `cookie`: cookie parameters that will be set in the client. Because our development environment doesn't have HTTPs, we turned off `secure`. In a production environment, the cookie should only be sent in secure contexts so secure should be true.
```

Comment:
what are cookie parameters -- its not explained. also cookies havent been explained to this point if im not mistaken

Reply:
Because I wasn't intending to talk about cookies as data holders. rather, simply their tied usage to session identifiers. I think another topic can be devoted to talking about cookies in general and their parameters.